editor/src/components/inspector/common/longhand-shorthand-hooks.ts

```diff
@@ -0,0 +1,236 @@
+import * as deepEqual from 'fast-deep-equal'
+import { useContextSelector } from 'use-context-selector'
+import { flatMapArray, last } from '../../../core/shared/array-utils'
+import { jsxAttributeValue } from '../../../core/shared/element-template'
+import { create } from '../../../core/shared/property-path'
+import { instancePath, isInstancePath } from '../../../core/shared/template-path'
+import { arrayEquals } from '../../../core/shared/utils'
+import { emptyComments } from '../../../core/workers/parser-printer/parser-printer-comments'
+import { useKeepReferenceEqualityIfPossible } from '../../../utils/react-performance'
+import {
+  getControlStatusFromPropertyStatus,
+  getControlStyles,
+  PropertyStatus,
+} from '../../../uuiui-deps'
+import {
+  setProp_UNSAFE,
+  transientActions,
+  unsetProperty,
+} from '../../editor/actions/action-creators'
+import { useEditorState } from '../../editor/store/store-hook'
+import { ParsedProperties, ParsedPropertiesKeys, printCSSValue } from './css-utils'
+import {
+  InspectorInfo,
+  InspectorPropsContext,
+  ParsedValues,
+  PathMappingFn,
+  useGetOrderedPropertyKeys,
+  useInspectorContext,
+  useInspectorInfo,
+} from './property-path-hooks'
+
+function getShadowedLonghandShorthandValue<
+  LonghandKey extends ParsedPropertiesKeys,
+  ShorthandKey extends ParsedPropertiesKeys
+>(
+  longhand: LonghandKey,
+  shorthand: ShorthandKey,
+  longhandPropertyStatus: PropertyStatus,
+  shorthandPropertyStatus: PropertyStatus,
+  longhandValue: ParsedProperties[LonghandKey],
+  shorthandValue: ParsedProperties[ShorthandKey],
+  orderedPropKeys: (LonghandKey | ShorthandKey)[][], // multiselect
+): { value: ParsedProperties[LonghandKey]; propertyStatus: PropertyStatus } {
+  const allPropKeysEqual = orderedPropKeys.every((propKeys) => {
+    return arrayEquals(propKeys, orderedPropKeys[0])
+  })
+
+  const propKeys = orderedPropKeys[0] ?? []
+  const lastKey = last(propKeys)
+  if (lastKey == null) {
+    return {
+      value: longhandValue,
+      propertyStatus: longhandPropertyStatus,
+    }
+  } else {
+    if (lastKey === longhand) {
+      return {
+        value: longhandValue,
+        propertyStatus: allPropKeysEqual
+          ? longhandPropertyStatus
+          : {
+              ...longhandPropertyStatus,
+              set: true,
+              controlled: true,
+              overwritable: false,
+              identical: false,
+            },
+      }
+    } else {
+      // Important: we ASSUME that the transformed shorthand value is an object,
+      // where the keys are the longhand keys and the values are the individual longhand values
```

Comment:
By "transformed shorthand value" here do you mean `shorthandValue`, i.e. the param to this function? It looks like you do, in which case please can you rename that param to something that hints that it should be an object that is the same shape as the longhand value (and move this comment to the function signature)

Reply:
I left this comment here (with some modification) but also fixed the variable name and left a smaller comment there

Reply:
Works for me! Thanks!